File: coding_freshmen/JAVA/Soumyaranjan/MAJORITY_ELEMENT.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Size of the array: ");
        int l = sc.nextInt();
        int[] nums = new int[l];
        for (int i = 0; i < l; i++) {
            nums[i] = sc.nextInt();
        }

        System.out.println(majorityElement(nums));
    }
    static int majorityElement(int[] arr){
        int count = 0, max = 0, temp = -1;
        for (int j : arr) {
            for (int k : arr)
                if (j == k) {
                    count++;
                    if (count > max) {
                        max = count;
                        temp = j;
                    }
                }
            count = 0;
        }
        return temp;
    }
}

File: coding_freshmen/C++/TANIKA_GULATI/MAJORITY_ELEMENT.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n;
    cin >> n;
    int nums[n];
    for (int i = 0; i < n; i++)
    {
        cin >> nums[i];
    }
    int max;
    int count = 0;
    for (int i = 0; i < n; i++)
    {
        if (count == 0)
        {
            max = nums[i];
            count = 1;
            continue;
        }
        if (max != nums[i])
        {
            count--;
        }
        else
        {
            count++;
        }
    }
    cout << max;
}
